templates/observability-remote-write-proxy.yml

```diff
@@ -0,0 +1,335 @@
+  - kind: Service
+    apiVersion: v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy-metrics
+        app.kubernetes.io/name: observability-remote-write-proxy-metrics
+      name: observability-rw-proxy-metrics
+    spec:
+      ports:
+        - name: metrics
+          port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          targetPort: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_METRICS_PORT}}
+          protocol: TCP
+      selector:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: PodDisruptionBudget
+    apiVersion: policy/v1
+    metadata:
+      name: observability-remote-write-proxy-pdb
+    spec:
+      minAvailable: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PDB_MIN_AVAILABLE}}
+      selector:
+        matchLabels:
+          app.kubernetes.io/name: observability-remote-write-proxy
+  - kind: NetworkPolicy
+    apiVersion: networking.k8s.io/v1
+    metadata:
+      labels:
+        app.kubernetes.io/component: prometheus-remote-write-proxy
+        app.kubernetes.io/name: observability-remote-write-proxy
+      name: observability-remote-write-proxy
+    spec:
+      podSelector:
+        matchLabels:
+          app.kubernetes.io/component: prometheus-remote-write-proxy
+          app.kubernetes.io/name: observability-remote-write-proxy
+      policyTypes:
+      - Ingress
+      ingress:
+      - from:
+          # Allow access from the default OpenShift Ingress Controller pods.
+          # The default OpenShift Ingress Controller pods are located in a K8s
+          # Namespace that contains the 'network.openshift.io/policy-group' K8s
+          # label key with the value 'ingress' and the 'name' label with the
+          # value 'openshift-ingress'. Additionally, those pods have the label
+          # 'ingresscontroller.operator.openshift.io/deployment-ingresscontroller'
+          # with the value 'default'.
+        - namespaceSelector:
+            matchLabels:
+              network.openshift.io/policy-group: ingress
+              name: openshift-ingress
+          podSelector:
+            matchLabels:
+              ingresscontroller.operator.openshift.io/deployment-ingresscontroller: default
+        ports:
+        - port: ${{OBSERVABILITY_REMOTE_WRITE_PROXY_PORT}}
+          protocol: TCP
+
+parameters:
+- name: OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_NAME
+  description: Observability Remote Write Proxy container image name
+  value: "quay.io/rhoas/observability-remote-write-proxy"
+
+- name: OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_TAG
+  description: Observability Remote Write Proxy container image tag
+  value: "latest"
+
+- name: OBSERVABILITY_REMOTE_WRITE_PROXY_IMAGE_PULL_POLICY
+  description: >-
+    Observability Remote Write Proxy container image pull policy.
+    Accepted values: ['Always', 'IfNotPresent', 'Never']"
+  value: "IfNotPresent"
```

Comment:
Two options here:
Set the default as `Always` and in AppInterface set it explicitly to IfNotPresent. Or leave it as IfNotPresent in which case during development in some cases it will be needed to be changed (for example if a `latest` tag is used)

Reply:
+1 to set `Always` as the default, this allows us to use latest.

Reply:
Ok. But then in Stage/Prod we don't want `Always` and we will need to override there. The reason for that is that there we will use a fixed tag and it is unnecessary to pull the image every time